docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,595 @@
+    spec:
+        namespaceSelector:
+          matchExpressions:
+          - key: kubernetes.io/metadata.name
+            operator: In
+            values: ["ns1", "ns2"]
+        network:
+          topology: Layer3
+          layer3:
+            role: Primary
+            # The UDN MTU shall not be larger than the provider network's MTU.
+            mtu: 1500
+            subnets:
+            - cidr: 10.10.0.0/16
+              hostSubnet: 24
+            encapsulation: None
+    ```
+
+1. The user has created an FRRConfiguration CR to peer with a external BGP
+   router `182.18.0.5` on the blue VLANs.
+
+    ```yaml
+    apiVersion: frrk8s.metallb.io/v1beta1
+    kind: FRRConfiguration
+    metadata:
+      name: vpn-blue
+      namespace: frr-k8s-system
+      labels:
+        network: vpn-blue
+    spec:
+      bgp:
+        routers:
+        - asn: 64512
+          vrf: blue
+          neighbors:
+          - address: 182.18.0.5
+            asn: 64512
+            disableMP: true
+            holdTime: 1m30s
+            keepaliveTime: 30s
+            passwordSecret: {}
+            port: 179
+            toAdvertise:
+              allowed:
+                mode: filtered
+            toReceive:
+              allowed:
+                mode: filtered
+    ```
+
+1. The user have to advertise this pod network of the CUDN by BGP. The targetVRF
+   is set to auto, meaning routes to the podNetworks will only be advertised
+   within the corresponding VRF.
+
+    ```yaml
+    apiVersion: k8s.ovn.org/v1
+    kind: RouteAdvertisements
+    metadata:
+      name: blue
+    spec:
+      # advertise routes to target VRF vpn-blue
+      targetVRF: auto
+      # nodeSelector must be empty, since we don't support a network in a overlay and no-overlay hybrid mode.
+      nodeSelector: {}
+      frrConfigurationSelector:
+        network: vpn-blue
+      networkSelectors:
+      - networkSelectionType: ClusterUserDefineNetwork
+        clusterUserDefinedNetworkSelector:
+          networkSelector:
+            matchLabels:
+              # Select one CUDN to ensure the cross UDN isolation
+              advertise: blue
+      advertisements:
+      - PodNetwork
+    ```
+
+1. OVN-Kubernetes will generate a following FRRConfiguration for each node.
+
+    ```yaml
+    apiVersion: frrk8s.metallb.io/v1beta1
+    kind: FRRConfiguration
+    metadata:
+      name: route-generated-blue
+      namespace: frr-k8s-system
+    spec:
+      bgp:
+        routers:
+        - asn: 64512
+          neighbors:
+          - address: 182.18.0.5
+            disableMP: true
+            asn: 64512
+            toAdvertise:
+              allowed:
+                prefixes:
+                # the node subnet of this node
+                - 10.10.1.0/24
+          vrf: blue
+          prefixes:
+          - 10.0.1.0/24
+    ```
+
+### Deployment Consideration
+
+#### BGP Topology
+
+The deployment mode will be decided by the FRRConfiguration CR that is selected
```

Comment:
To stress that it's up to the cluster admin:
```suggestion
The deployment mode will be decided by the FRRConfiguration CR that the user creates
```

BTW should we replace `user` with `cluster admin` all throughout the enhancement?